lib/git.js

```diff
@@ -1,6 +1,6 @@
 exports.log = function (options, callback) {
 	var spawn = require("child_process").spawn;
-	var gitArgs = ["log", "--no-color", "--no-merges", "--branches=" + options.branch, "--format=" + formatOptions, options.range];
+	var gitArgs = ["log", "--no-color", "--no-merges", "--extended-regexp", "--branches=" + options.branch, "--format=" + formatOptions, "--grep=" + options.grep, options.range];
```

Comment:
I would rather add the arguments `--extended-regexp` and `--grep` __only if__ `options.grep` is not void (understand not `null`, not `undefined`) nor empty.

Something like: 

```javascript
var grep = options.grep;
var hasRegExp = grep != null && grep.length > 0;
var gitArgs = ["log", "--no-color", "--no-merges"];
if (hasRegExp) {
    gitArgs.push("--extended-regexp");
}
gitArgs = gitArgs.concat(["--branches=" + options.branch, "--format=" + formatOptions]);
if (hasRegExp) {
    gitArgs.push("--grep=" + grep);
}
gitArgs = gitArgs.concat([options.range]);
```

I kept the last line as a `concat` for maintainability purpose. Also, if the two arguments related to the regular expression can be put next to each other, this would avoid having to check twice and to cache the result of the check as I did, thus simplifying the code.